priv/repo/migrations/20180628175114_create_user_impressions_materialized_view_v2.exs

```diff
@@ -0,0 +1,39 @@
+defmodule CodeFund.Repo.Migrations.CreateUserImpressionsMaterializedViewV2 do
+  use Ecto.Migration
+
+  def change do
+    execute("DROP MATERIALIZED VIEW user_impressions CASCADE")
+
+    execute("""
+      CREATE MATERIALIZED VIEW user_impressions AS
+        SELECT campaigns.user_id AS campaign_user_id,
+        impressions.id,
+        impressions.campaign_id,
+        impressions.revenue_amount,
+        impressions.distribution_amount,
+        impressions.inserted_at,
+        properties.name AS property_name,
+        properties.user_id AS property_user_id,
+        audiences.name AS audience_name,
+        campaigns.name AS campaign_name,
+        users.company AS advertiser_company_name
+        FROM impressions
+        JOIN campaigns ON impressions.campaign_id = campaigns.id
+        JOIN properties ON impressions.property_id = properties.id
+        JOIN audiences ON campaigns.audience_id = audiences.id
+        JOIN users ON campaigns.user_id = users.id;
+    """)
+
+    create(unique_index("user_impressions", [:id]))
+
+    create(index("user_impressions", [:campaign_user_id]))
```

Comment:
I wonder if some of these wouldn't benefit from being compound indices depending on how you're planning to query them?